packages/react-components/src/components/Datepicker/helpers.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  isAfter,
+  isSameDay,
+  isSameMonth,
+  subMonths,
+  differenceInCalendarDays,
+} from 'date-fns';
+import { Modifiers } from 'react-day-picker';
+import {
+  IRangeDatePickerProps,
+  IRangeDatePickerState,
+  IRangeDatePickerOption,
+} from './types';
+
+const baseClass = 'date-picker';
+
+import styles from './DatePicker.module.scss';
+
+export const isDateWithinRange = (
+  date: Date,
+  range: { from?: Date; to?: Date }
+): boolean => {
+  const { from, to } = range;
+  if (to && !isSameDay(date, to) && isAfter(date, to)) {
+    return false;
+  }
+  if (from && !isSameDay(date, from) && !isAfter(date, from)) {
+    return false;
+  }
+  return true;
+};
+
+export const calculateDatePickerMonth = (
+  initialFromDate?: Date,
+  initialToDate?: Date,
+  toMonth?: Date
+): Date => {
+  if (initialToDate) {
+    const forcePreviousMonth =
+      initialFromDate && !isSameMonth(initialFromDate, initialToDate);
+
+    if (
+      forcePreviousMonth ||
+      (toMonth && isSameMonth(initialToDate, toMonth))
+    ) {
+      return subMonths(initialToDate, 1);
+    }
+
+    return initialToDate;
+  }
+
+  return subMonths(toMonth || new Date(), 1);
+};
+//styles[`${baseClass}__day--start`]
+
+export const getRangeDatePickerModifiers = (
+  from?: Date,
+  to?: Date
+): Partial<Modifiers> => {
+  const base = {
+    [styles[`${baseClass}__day--monday`]]: { daysOfWeek: [1] },
+    [styles[`${baseClass}__day--sunday`]]: { daysOfWeek: [0] },
+    [styles[`${baseClass}__day--start`]]: from,
+    [styles[`${baseClass}__day--end`]]: from,
+  };
+
+  if (to) {
+    if (!from) {
+      return {
+        ...base,
+        [styles[`${baseClass}__day--start`]]: to,
+      };
+    }
+
+    const diff = differenceInCalendarDays(to, from);
+
+    if (diff > 0) {
+      return {
+        ...base,
+        [styles[`${baseClass}__day--end`]]: to,
+      };
+    } else if (diff < 0) {
+      return {
+        ...base,
+        [styles[`${baseClass}__day--start`]]: to,
+      };
+    }
+    return {
+      ...base,
+      [styles[`${baseClass}__day--start`]]: [from, to],
+      [styles[`${baseClass}__day--end`]]: [from, to],
+    };
+  }
```

Comment:
It would be possible to make these `if` statements flatter

Reply:
☝️

Reply:
this code has been "inherited" from the last version of DS. I have simplified the code here, and seems to be working, but maybe I have missed some use cases here.